### PR TITLE
Use one set of worker queues for timer processing

### PR DESCRIPTION
**Pull request by shreyassrivatsan**
Pull out worker queue logic - initialize and use in timer queue implementations.

### CODE REVIEW

File: service/history/timerQueueTaskProcessor.go [hunk 1]
```diff
@@ -0,0 +1,322 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
```
Comment (wxing1292) [inline service/history/timerQueueTaskProcessor.go]: plz also port the existing tests to `timerQueueTaskProcessor_test.go`

File: service/history/timerQueueProcessorBase_test.go [hunk 2]
```diff
@@ -66,7 +66,8 @@ type (
 		notificationChan chan struct{}
```
Comment (wxing1292) [inline service/history/timerQueueProcessorBase_test.go]: FYI timerQueueProcessor_test.go and timerQueueProcessor2_test.go needs rewrite (to become a unit test, not integration test which includes db)

File: service/history/timerQueueTaskProcessor.go (hunk 1 above)
Comment (shreyassrivatsan) [reply service/history/timerQueueTaskProcessor.go]: done

File: service/history/timerQueueProcessorBase_test.go (hunk 2 above)
Comment (shreyassrivatsan) [reply service/history/timerQueueProcessorBase_test.go]: yes, I saw that.. will change as part of future diffs..

**Pull request merged**